libs/community/langchain_community/tools/azure_cognitive_services/image_analysis.py

```diff
@@ -14,6 +15,11 @@
 
 logger = logging.getLogger(__name__)
 
+warnings.warn(
+    "The 'azure-ai-vision' package has been yanked from PyPI and replaced by 'azure-ai-vision-imageanalysis'. "
```

Comment:
should we update the code to use azure-ai-vision-imageanalysis? or are the packages sufficiently different that we'd basically need to rewrite the integration

Reply:
Unfortunately, `azure-ai-vision-imageanalysis` is not a drop-in replacement; The tool would need to be rewritten. [the SDK has been rewritten and all APIs have changed](https://learn.microsoft.com/en-us/azure/ai-services/computer-vision/whats-new#new-image-analysis-sdk-100-beta1-breaking-changes).

I did a deeper dive here to see what the differences are:
- The .whl files provided by `azure-ai-vision` and `azure-ai-vision-imageanalysis` are not even similar when analyzed.
- I also took the time to print out and analyze [this commit](https://github.com/MicrosoftDocs/azure-docs/commit/943b431c36fbc0df0e8a501e0c9ce3b996367cf2?diff=split&w=0) where they update the documentation from `azure-ai-vision` to `azure-ai-vision-imageanalysis` and there are differences in how you load local images (you now load the image as bytes instead of creating a `VisionSource` object). Additionally, they way of getting results from the service, error handling, and logging are all significantly different.